bin/reth/src/node/mod.rs

```diff
@@ -259,6 +272,32 @@ impl Command {
                 .build(fetch_client.clone(), consensus.clone(), db.clone()),
         )
     }
+
+    async fn load_peers(&self, network: &NetworkHandle) -> Result<(), eyre::Error> {
+        let reader = match std::fs::File::open(&self.known) {
+            Ok(file) => std::io::BufReader::new(file),
+            Err(e) if e.kind() == ErrorKind::NotFound => return Ok(()),
+            Err(e) => Err(e)?,
+        };
+        info!(target: "reth::cli", file = ?self.known, "Loading saved peers");
+        let known_peers: Vec<NodeRecord> = serde_json::from_reader(reader)?;
+        known_peers.into_iter().for_each(|record| network.add_peer(record.id, record.udp_addr()));
+        Ok(())
+    }
+
+    // Dumps peers to `file_path` for persistence.
+    async fn dump_peers(&self, network: &NetworkHandle) -> Result<(), eyre::Error> {
+        info!(target: "reth::cli", file = ?self.known, "Saving current peers");
+        let writer = std::io::BufWriter::new(std::fs::File::create(&self.known)?);
+        let known_peers: Vec<NodeRecord> = network
+            .get_peers()
+            .await?
+            .into_iter()
+            .map(|PeerInfo { remote_id, remote_addr, .. }| NodeRecord::new(remote_addr, remote_id))
+            .collect();
+        serde_json::to_writer_pretty(writer, &known_peers)?;
```

Comment:
Can we make this 
1. opt-out
2. store the reputation of the peers?
3. only save peers above x% reputation based on the max?

Reply:
> store the reputation of the peers?

I don't think this is necessary, because the peers here don't include peers that we know are useless (different network for example) and reputation is merely used to rank peers or put them in timeout if they are busy at the moment, which is perhaps no longer the case when the node is restarting.

Reply:
Added a cli flag for disabling peer persistence. Also added peer filtering based on reputation. For now it saves only peers with reputation higher than max peer reputation minus `10000`